internal/repository/payment_method_repository.go

```diff
@@ -0,0 +1,111 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tubagusmf/payment-service-gb1/internal/model"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type PaymentRepository struct {
+	db    *sql.DB
+	redis *redis.Client
+}
+
+func NewPaymentRepository(db *sql.DB, redis *redis.Client) model.IPaymentRepository {
+	return &PaymentRepository{
+		db:    db,
+		redis: redis,
+	}
+}
+
+func (p *PaymentRepository) FindAll(ctx context.Context, filter model.PaymentFilter) ([]*model.Payment, error) {
+
+	res, err := p.db.QueryContext(ctx, "SELECT id, name, bank_code, created_at, updated_at, deleted_at FROM payment_methods LIMIT ? OFFSET ?", filter.Limit, filter.Offset)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var payments []*model.Payment
+	for res.Next() {
+		var payment model.Payment
+		if err := res.Scan(&payment.Id, &payment.Name, &payment.BankCode, &payment.CreatedAt, &payment.UpdatedAt, &payment.DeletedAt); err != nil {
+			return nil, err
+		}
+		payments = append(payments, &payment)
+	}
+
+	return payments, nil
+}
+
+func (p *PaymentRepository) FindById(ctx context.Context, id int64) (*model.Payment, error) {
+	paymentKey := getPaymentKey(id)
+
+	var payment model.Payment
+	pm, err := p.redis.Get(ctx, paymentKey).Result()
+	if err == nil {
+		err := json.Unmarshal([]byte(pm), &payment)
+		if err != nil {
+			return nil, err
+		}
+		return &payment, nil
+	}
+
+	res, err := p.db.QueryContext(ctx, "SELECT id, name, bank_code, created_at, updated_at, deleted_at FROM payment_methods WHERE id=?", id)
+
+	if err != nil {
+		return nil, err
```

Comment:
mas sekalian cek kalo error not found jangan return error tapi nil aja. nanti kalo hasil querynya nil maka return not found, biar gak jadi internal server error. Bisa pake ini
```
if err == sql.ErrNoRows {
   return nil, nil
}
```